### ISSUE TITLE
Can't build paru 1.7.1 on armv7 - alpm-2.0.2/src/types.rs error[E0308]: mismatched types

### ISSUE THREAD

**Issue by mrdotx** [Morganamilo/paru#402]
### Affected Version

paru v1.7.1
Arch Linux armv7l

### Description

On Arch ARM updating paru from v1.6.1 to v1.7.1, compilation breaks up with a few 'mismatched types' errors on alpm-2.0.2/src/types.rs.

- pacman v6.0.0 is installed

### Output

```sh
[share/repos/paru master=]» makepkg -si                                                                                                                                          
==> Erstelle Paket: paru 1.7.1-1 (Do 03 Jun 2021 07:59:55 CEST)
==> Prüfe Laufzeit-Abhängigkeiten...
==> Prüfe Buildtime-Abhängigkeiten...
==> Empfange Quellen...
  -> Lade paru-1.7.1.tar.gz herunter...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   124  100   124    0     0    502      0 --:--:-- --:--:-- --:--:--   504
100 99207  100 99207    0     0   138k      0 --:--:-- --:--:-- --:--:-- 1335k
==> Überprüfe source Dateien mit sha256sums...
    paru-1.7.1.tar.gz ... Durchgelaufen
==> Entpacke Quellen...
  -> Entpacke paru-1.7.1.tar.gz mit bsdtar
==> Beginne build()...
   Compiling libc v0.2.95
   Compiling proc-macro2 v1.0.27
   Compiling unicode-xid v0.2.2
   Compiling autocfg v1.0.1
   Compiling syn v1.0.72
   Compiling cfg-if v1.0.0
   Compiling memchr v2.4.0
   Compiling log v0.4.14
   Compiling proc-macro-hack v0.5.19
   Compiling getrandom v0.1.16
   Compiling ppv-lite86 v0.2.10
   Compiling pin-project-lite v0.2.6
   Compiling futures-core v0.3.15
   Compiling siphasher v0.3.5
   Compiling bitflags v1.2.1
   Compiling lazy_static v1.4.0
   Compiling bytes v1.0.1
   Compiling once_cell v1.7.2
   Compiling itoa v0.4.7
   Compiling proc-macro-nested v0.1.7
   Compiling futures-sink v0.3.15
   Compiling matches v0.1.8
   Compiling futures-channel v0.3.15
   Compiling cc v1.0.68
   Compiling pkg-config v0.3.19
   Compiling serde_derive v1.0.126
   Compiling futures-task v0.3.15
   Compiling serde v1.0.126
   Compiling futures-io v0.3.15
   Compiling slab v0.4.3
   Compiling pin-utils v0.1.0
   Compiling fnv v1.0.7
   Compiling alpm-sys v2.0.0
   Compiling tinyvec_macros v0.1.0
   Compiling foreign-types-shared v0.1.1
   Compiling openssl v0.10.34
   Compiling ryu v1.0.5
   Compiling encoding_rs v0.8.28
   Compiling native-tls v0.2.7
   Compiling hashbrown v0.9.1
   Compiling percent-encoding v2.1.0
   Compiling crc32fast v1.2.1
   Compiling httparse v1.4.1
   Compiling openssl-probe v0.1.4
   Compiling try-lock v0.2.3
   Compiling new_debug_unreachable v1.0.4
   Compiling adler v1.0.2
   Compiling mac v0.1.1
   Compiling serde_json v1.0.64
   Compiling httpdate v1.0.1
   Compiling tower-service v0.3.1
   Compiling precomputed-hash v0.1.1
   Compiling dtoa v0.4.8
   Compiling utf-8 v0.7.6
   Compiling async-trait v0.1.50
   Compiling regex-syntax v0.6.25
   Compiling ipnet v2.3.0
   Compiling smallvec v1.6.1
   Compiling byteorder v1.4.3
   Compiling convert_case v0.4.0
   Compiling cini v0.1.1
   Compiling mime v0.3.16
   Compiling base64 v0.13.0
   Compiling stable_deref_trait v1.2.0
   Compiling nodrop v0.1.14
   Compiling thin-slice v0.1.1
   Compiling anyhow v1.0.40
   Compiling remove_dir_all v0.5.3
   Compiling termcolor v1.1.2
   Compiling humantime v2.1.0
   Compiling number_prefix v0.4.0
   Compiling ansi_term v0.12.1
   Compiling srcinfo v1.0.0
   Compiling htmlescape v0.3.1
   Compiling futures-macro v0.3.15
   Compiling tokio v1.6.1
   Compiling futures-util v0.3.15
   Compiling indexmap v1.6.2
   Compiling miniz_oxide v0.4.4
   Compiling num-traits v0.2.14
   Compiling num-integer v0.1.44
   Compiling tracing-core v0.1.18
   Compiling phf_shared v0.8.0
   Compiling unicode-bidi v0.3.5
   Compiling http v0.2.4
   Compiling tinyvec v1.2.0
   Compiling foreign-types v0.3.2
   Compiling form_urlencoded v1.0.1
   Compiling openssl-sys v0.9.63
   Compiling futf v0.1.4
   Compiling dtoa-short v0.3.3
   Compiling fxhash v0.2.1
   Compiling pacmanconf v1.0.0
   Compiling servo_arc v0.1.1
   Compiling tracing v0.1.26
   Compiling unicode-normalization v0.1.18
   Compiling http-body v0.4.2
   Compiling tendril v0.4.2
   Compiling quote v1.0.9
   Compiling want v0.3.0
   Compiling aho-corasick v0.7.18
   Compiling bstr v0.2.16
   Compiling signal-hook-registry v1.3.0
   Compiling num_cpus v1.13.0
   Compiling mio v0.7.11
   Compiling getrandom v0.2.3
   Compiling socket2 v0.4.0
   Compiling terminal_size v0.1.17
   Compiling atty v0.2.14
   Compiling dirs-sys v0.3.6
   Compiling time v0.1.44
   Compiling term_size v0.3.2
   Compiling nix v0.20.0
   Compiling alpm v2.0.2
   Compiling idna v0.2.3
   Compiling quick-xml v0.20.0
   Compiling regex v1.5.4
   Compiling rand_core v0.6.2
   Compiling console v0.14.1
   Compiling dirs v3.0.2
   Compiling rand_core v0.5.1
   Compiling flate2 v1.0.20
   Compiling url v2.2.2
   Compiling rss v1.10.0
   Compiling rand_chacha v0.3.0
   Compiling indicatif v0.16.2
   Compiling env_logger v0.8.3
   Compiling globset v0.4.6
   Compiling rand_pcg v0.2.1
   Compiling rand_chacha v0.2.2
   Compiling chrono v0.4.19
error[E0308]: mismatched types
   --> /home/alarm/.cargo/registry/src/github.com-1285ae84e5963aae/alpm-2.0.2/src/types.rs:584:18
    |
583 |     pub fn total_size(&self) -> i64 {
    |                                 --- expected `i64` because of return type
584 |         unsafe { (*self.inner).total_size }
    |                  ^^^^^^^^^^^^^^^^^^^^^^^^
    |                  |
    |                  expected `i64`, found `i32`
    |                  help: you can convert an `i32` to an `i64`: `(*self.inner).total_size.into()`

error[E0308]: mismatched types
    --> /home/alarm/.cargo/registry/src/github.com-1285ae84e5963aae/alpm-2.0.2/src/types.rs:1107:42
     |
1107 |                     downloaded: unsafe { (*data).downloaded },
     |                                          ^^^^^^^^^^^^^^^^^^
     |                                          |
     |                                          expected `i64`, found `i32`
     |                                          help: you can convert an `i32` to an `i64`: `(*data).downloaded.into()`

error[E0308]: mismatched types
    --> /home/alarm/.cargo/registry/src/github.com-1285ae84e5963aae/alpm-2.0.2/src/types.rs:1108:37
     |
1108 |                     total: unsafe { (*data).total },
     |                                     ^^^^^^^^^^^^^
     |                                     |
     |                                     expected `i64`, found `i32`
     |                                     help: you can convert an `i32` to an `i64`: `(*data).total.into()`

error[E0308]: mismatched types
    --> /home/alarm/.cargo/registry/src/github.com-1285ae84e5963aae/alpm-2.0.2/src/types.rs:1127:37
     |
1127 |                     total: unsafe { (*data).total },
     |                                     ^^^^^^^^^^^^^
     |                                     |
     |                                     expected `i64`, found `i32`
     |                                     help: you can convert an `i32` to an `i64`: `(*data).total.into()`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0308`.
error: could not compile `alpm`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
==> FEHLER: Ein Fehler geschah in build().
    Breche ab...
```

**Comment by Morganamilo** [comment]
Should be fixed now with alpm 2.0.3.

**Comment by mrdotx** [comment]
I compiled the latest version (paru-git 1.7.1.r1.g9a211ed-1) and it works as expected again. Thanks!
